torch/_inductor/codegen/cuda/cuda_template.py

```diff
@@ -78,26 +83,18 @@ def generate(self, **kwargs) -> CUDATemplateCaller:
             extra_args=extra_args,
             source_code=code,
         )
-
-        def make_kernel_render(output_node):
-            kernel = CUDATemplateKernel(
-                kernel_name="KERNEL_NAME",
-            )
-            render = functools.partial(
-                self.render,
-                kernel=kernel,
-                output_node=output_node,
-                **kwargs,
-            )
-            return kernel, render
+        cuda_template_buffer = CUDATemplateBuffer(
```

Comment:
This had to be changed quite a bit. These are the closures which hid all the information I was trying to get at. So instead of a closure ( which is just an implicit private callable object, really ), I'm passing around the information explicitly within CUDATemplateBuffer. The only thing that still requires a callback is the workspace_size, as that's not known at this point yet.